### PR TITLE
Kaisong1990/add time interval option

### PR DESCRIPTION
**Pull request by kaisong1990**
## Proposed changes

Expose APIs for manually setting time_Interval for request and session resource

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by kaisong1990** [comment]
Broker PR is here:
https://github.com/AzureAD/azure-activedirectory-tokenbroker-for-objc/pull/1352#pullrequestreview-1596630784

File: IdentityCore/src/network/MSIDURLSessionManager.m [hunk 1]
```diff
@@ -25,14 +25,38 @@
...
+
+- (void)dealloc
 {
-    if (self == [MSIDURLSessionManager self])
+    [_session invalidateAndCancel];
+}
+
++ (MSIDURLSessionManager *)defaultManager
+{
+    if (!s_defaultManager)
```
Comment (oldalton) [inline IdentityCore/src/network/MSIDURLSessionManager.m:53]: Should this also be wrapped in dispatch once? Or synchronized?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/network/MSIDURLSessionManager.m (hunk 1 above)
Comment (kaisong1990) [reply IdentityCore/src/network/MSIDURLSessionManager.m]: Updated with dispatch once

**Comment by fengga** [comment]
Thanks, I'll apply the same changes in MSAL CPP.

File: IdentityCore/src/network/MSIDURLSessionManager.m [hunk 2]
```diff
@@ -25,14 +25,38 @@
 #import "MSIDURLSessionDelegate.h"
 
 static MSIDURLSessionManager *s_defaultManager = nil;
+static NSTimeInterval s_timeoutIntervalForResource = 0;
 
 @implementation MSIDURLSessionManager
 
-+ (void)initialize
+- (instancetype)initWithConfiguration:(NSURLSessionConfiguration *)configuration
```
Comment (fengga) [inline IdentityCore/src/network/MSIDURLSessionManager.m:32]: Kai, can you hold this PR for some time? I'm considering let MSAL CPP use this class, which may need some other changes. So MSAL CPP will not need to have its own implementation.

Comment (kaisong1990) [reply IdentityCore/src/network/MSIDURLSessionManager.m:32]: Synced offline, for combining CPP and CommonCore logic, we will have a separate item and not rush into the coming Sept hotfix

**Comment by kaisong1990** [comment]
The `[Broker] Common core submodule update check` is previously succeeded already, merging

https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1162628&view=results

**Pull request merged**